pkg/custompluginmonitor/custom_plugin_monitor.go

```diff
@@ -263,12 +265,17 @@ func (c *customPluginMonitor) generateStatus(result cpmtypes.Result) *types.Stat
 			}
 		}
 	}
-	return &types.Status{
+	status := &types.Status{
 		Source: c.config.Source,
 		// TODO(random-liu): Aggregate events and conditions and then do periodically report.
 		Events:     append(activeProblemEvents, inactiveProblemEvents...),
 		Conditions: c.conditions,
 	}
+	// Log only if condition has changed for permanent error
```

Comment:
Can we check whether `activeProblemEvents` and `inactiveProblemEvents` are empty? I think that should cover all cases.

Reply:
Switched to checking for active and inactive problems. They so cover all the cases.